app/controllers/api/trainees/degrees_controller.rb

```diff
@@ -13,33 +13,28 @@ def index
       end
 
       def create
-        trainee = current_provider.trainees.find_by!(slug: params[:trainee_slug])
-        degree_attributes = degree_attributes_service.new(degree_params)
-
         render(
-          Api::CreateDegree.call(trainee:, degree_attributes:, current_version:),
+          **SaveDegreeResponse.call(
+            degree: new_degree,
+            params: degree_params,
+            version: current_version,
+          ),
         )
       end
 
       def update
-        trainee = current_provider.trainees.find_by!(slug: params[:trainee_slug])
-        degree = trainee.degrees.find_by!(slug: params[:slug])
-
-        begin
-          attributes = degree_attributes_service.from_degree(degree)
-          attributes.assign_attributes(degree_update_params)
-          succeeded, errors = update_degree_service_class.call(degree:, attributes:)
-          if succeeded
-            render(json: { data: degree_serializer_class.new(degree).as_hash })
-          else
-            render(json: { errors: }, status: :unprocessable_entity)
-          end
-        rescue ActionController::ParameterMissing
-          render(
-            json: { errors: ["Request could not be parsed"] },
-            status: :unprocessable_entity,
-          )
-        end
+        render(
+          **SaveDegreeResponse.call(
+            degree: degree,
+            params: degree_params,
+            version: current_version,
+          ),
+        )
+      rescue ActionController::ParameterMissing
+        render(
+          json: { errors: ["Request could not be parsed"] },
+          status: :unprocessable_entity,
+        )
```

Comment:
Can we do something like that please


https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/controllers/api/base_controller.rb#L9-L11